src/app/schedule-view/component.ts

```diff
@@ -1,70 +1,140 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, OnDestroy } from '@angular/core';
 import { Router, ActivatedRoute, Params } from '@angular/router';
-import { SchedulePeriod } from './schedule/scheduleperiod';
 import { YacsService } from '../services/yacs.service';
 import { Course } from '../course-list/course/course';
 import { Schedule} from './schedule/schedule';
+import { ScheduleEvent } from './scheduleevent/scheduleevent';
+import { SelectionService } from '../services/selection.service';
+import 'rxjs/Rx';
+import {Subject,Subscription} from 'rxjs/Rx';
+
 
-const SCHEDULE_TEST_DATA: Schedule[] = [
-  // must instantiate Schedule with new + constructor
-  // since it has getter properties that would otherwise have to be declared
-  // in this array
-  new Schedule(480, 1200, [
-      {
-        name: 'sdfsdf',
-        crn: 54,
-        instructor: 'Dr Sdfsdf',
-        day: 1,
-        startTime: 480,
-        endTime: 540
-      }
-    ]
-  )
-];
 
 @Component({
   selector: 'schedule-view',
   templateUrl: './component.html',
   styleUrls: ['./component.scss']
 })
 
-export class ScheduleViewComponent {
+export class ScheduleViewComponent implements OnInit, OnDestroy{
+
+  courses : Course[];
+  isLoaded : boolean = false;
+  courseSelections : Object[];
+  courseIds: Object[];
+  sections : Object[];
+  processedSchedules: Schedule[]  = [];
+  crns : string;
+  schedules: Schedule[];
+  start: number = 480; //8AM    
+  end: number = 1200; //8PM
+
+  private subscription;
 
-  courses : Course[] = [];
-  params: Object = {department_id: 62};
 
   constructor (
     
     private yacsService : YacsService,
-    private activatedRoute: ActivatedRoute) { }
+    private selectionService : SelectionService,
+    private activatedRoute: ActivatedRoute) { 
+
+    this.subscription = this.selectionService.subscribe(
+      msg => {this.courseSelections = this.selectionService.getSelections(); this.getCourses();});
+  }
+
+ ngOnDestroy(){
+    this.subscription.unsubscribe();
+  }
 
   getCourses () {
+    this.sections = [];
+    this.courses = [];
+    this.courseIds = [];
     let newParams : Object = {
       show_sections: true,
       show_periods: true
     };
-    // add show_sections and show_periods to params
-    Object.assign(newParams, this.params); // cannot directly modify params
-    this.yacsService
-        .get('courses', newParams)
+    let i = 0;
+    this.isLoaded = false;
+    let len = Object.keys(this.courseSelections).length;
+    for(let key of Object.keys(this.courseSelections)){
```

Comment:
I'm a little confused here... It looks like we are making two requests per section, one to the courses api and one to the sections api. I can't seem to find any requests to the schedules api or any schedules being generated.

We need to make one request (total) to the schedules api with the list of section ids, and one request (total) to the courses api with the list of section ids. The latter probably shouldn't be done here though.